### PR TITLE
Production Tweaks

### PR DESCRIPTION
**Pull request by nmagerko**
This pull request adds no new features. Instead, it prepares the codebase to be deployed in a production environment. 

Summary of Changes:

- [x] Added Dockerfile for building production images
- [x] Added logging utilities to keep statements easy to track in production
- [x] Added CloudWatch logging transport for use in production
- [x] Added auto-reloading and auto-relaunching to development startup scripts only

Summary of Fixes:
- [x] Added mocked database connection during testing to avoid connection errors (#50)
- [x] Fixed issues with `npm run` scripts/commmands on certain platforms excluding Windows (untracked)

There are a couple of other things that I'd like to check on before we merge, but these are relatively minor. As such, I think a code review can take place whenever people have time, and I'll point out the (hopefully small) portions that I change later.

### CODE REVIEW

**Comment by nmagerko** [comment]
One of the things I might do is get rid of the File transport on the logger in development (or at least add a Console transport as well). I like the idea of having a persistent log while developing, but it's also nice to have things printed to the CLI. 

If there are any hard opinions on that, let me know.

File: api.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ config.cwd = process.__dirname;
```
Comment (YashoSharma) [inline api.js]: Are we no longer changing the API initialization flow? Or are the changes in the startup scripts what you were talking about?

File: config/dev.config.template [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ###############################################################################
 #ENV AND PORTS
 NODE_ENV=development
-PROFILE=hackillinois-api
+HACKILLINOIS_ID=instance0
```
Comment (YashoSharma) [inline config/dev.config.template]: Is this change to track load balancing easier?

File: api.js (hunk 1 above)
Comment (nmagerko) [inline api.js]: Yeah we'll no longer need to change the initialization flow. I think if we keep everything to be initialized via environment variables, we won't need any strange async initialization.

**Comment by tommypacker** [comment]
So I'm not super familiar with a lot of the apis we're using here, but the code itself looks good

**Pull request merged**